pkg/k8sutil/k8sutil.go

```diff
@@ -81,3 +82,23 @@ func InitOperatorService() (*v1.Service, error) {
 	}
 	return service, nil
 }
+
+// ResourceExists returns true if the given resource kind exists
+// in the given api groupversion
+func ResourceExists(dc discovery.DiscoveryInterface, apiGroupVersion, kind string) (bool, error) {
+	apiLists, err := dc.ServerResources()
```

Comment:
Could use [`dc.ServerResourcesForGroupVersion(apiGroupVersion)`](https://godoc.org/k8s.io/client-go/discovery#ServerResourcesInterface) instead to save a loop.

Reply:
Actually, if we do that we produce an error when the resource does not exist, and this check is intended to avoid producing an error when for example ServiceMonitor resource is not registered.

Reply:
But that reminded me I should not return an error when it doesn't exist, so thanks! :)

Reply:
I haven't tried it myself, but you may be able to use `errors.IsNotFound(err)` to check for that case.

Reply:
We don't want to be producing errors that are expected or can be predicted, that way we would lose clarity of actual errors happening and we would just ignore all errors. So these errors would show up in metrics, the APIserver exposes metrics and events about them.